Snowflake.Data/Core/SFBaseResultSet.cs

```diff
@@ -24,134 +24,33 @@ abstract class SFBaseResultSet
 
         protected abstract string getObjectInternal(int columnIndex);
 
-        internal byte[] getBytes(int columnIndex)
+        internal T GetValue<T>(int columnIndex)
         {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (byte[])SFDataConverter.convertToCSharpVal(val, sfDataType, typeof(byte[]));
-        }
-
-        internal DateTime getDateTime(int columnIndex)
-        {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (DateTime)SFDataConverter.convertToCSharpVal(val, sfDataType, typeof(DateTime));
-        }
-
-        internal Decimal getDecimal(int columnIndex)
-        {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (Decimal)SFDataConverter.convertToCSharpVal(val, sfDataType, typeof(Decimal));
-        }
-        internal double getDouble(int columnIndex)
-        {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (double)SFDataConverter.convertToCSharpVal(val, sfDataType ,typeof(double));
-        }
-        internal float getFloat(int columnIndex)
-        {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (float)SFDataConverter.convertToCSharpVal(val, sfDataType, typeof(float));
-        }
-        internal Guid getGuid(int columnIndex)
-        {
-            string val = getObjectInternal(columnIndex);
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-            return (Guid)SFDataConverter.convertToCSharpVal(val, sfDataType, typeof(Guid));
+            return (T) GetValue(columnIndex);
         }
 
-        internal string getString(int columnIndex)
+        internal string GetString(int columnIndex)
         {
-            SFDataType sfDataType = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
-
-            switch(sfDataType)
+            var type = sfResultSetMetaData.getColumnTypeByIndex(columnIndex);
+            switch (type)
             {
                 case SFDataType.DATE:
-                    return SFDataConverter.toDateString(getDateTime(columnIndex), 
+                    var val = GetValue(columnIndex);
+                    if (val == null)
+                        return null;
+                    return SFDataConverter.toDateString((DateTime)val, 
                         sfResultSetMetaData.dateOutputFormat);
+                //TODO: Feels like were missing some implementations here, at least for time?

```

Comment:
Yes. Because snowflake provide a parameters like TIMESTAMP_NTZ_OUTPUT_FORMAT, which allowed user to specify output format (detailed in https://docs.snowflake.net/manuals/user-guide/date-time-input-output.html#output-formats). However, it requires some effort to correct implement parser and formatter of the snowflake timestamp format, I decide to defer the implementation when some users actually asked for it.